File: cinebook-frontend/src/services/ScreenService.js
import axiosInstance from "./axiosInstance";

export const addScreen=(screen)=>{
    return axiosInstance.post("/api/screen",screen);
}
export const getAllScreen=(page=0,size=5)=>{
    return axiosInstance.get(`/api/screen?page=${page}&size=${size}`);

}
export const allScreens=()=>{
    return axiosInstance.get("/api/screen/all");
}

export const getScreenByid=(id)=>{
    return axiosInstance.get(`/api/screen/${id}`);
}

export const updateScreen=(id,screen)=>{
    return axiosInstance.put(`/api/screen/${id}`,screen);
}

export const activateScreen=(id)=>{
    return axiosInstance.put(`/api/screen/${id}/activate`);
}
export const deactivateScreen=(id)=>{
    return axiosInstance.put(`/api/screen/${id}/deactivate`);
}

export const searchScreen=(screenName)=>{
    return axiosInstance.get(`/api/screen/search/${screenName}`);

}

export const searchByScreenType=(screenType)=>{
    return axiosInstance.get(`/api/screen/search/type/${screenType}`);
}

export const countActiveScreens=()=>{
    return axiosInstance.get("/api/screen/count/active");
}

export const getActiveScreens=()=>{
    return axiosInstance.get("/api/screen/active");
}

export const getNotActiveScreen=()=>{
    return axiosInstance.get("/api/screen/Inactive");
}

export const getScreensByTheater=(theaterId)=>{
    return axiosInstance.get(`/api/screen/theater/${theaterId}`);
}

File: cinebook-backend/cinebook-backend/src/main/java/com/cinebook/cinebook/service/ScreenService.java
package com.cinebook.cinebook.service;

import com.cinebook.cinebook.dto.request.ScreenRequestDto;
import com.cinebook.cinebook.dto.response.ScreenResponseDto;
import com.cinebook.cinebook.entity.Theater;
import org.springframework.data.domain.Page;

import java.util.List;

public interface ScreenService  {


    //add screen
    ScreenResponseDto addScreen(ScreenRequestDto requestDto);
    //get the screens with pagination
    Page<ScreenResponseDto> getAllScreen(int page,int size,String sortBy,String sortDir);
    List<ScreenResponseDto>allScreens();
    //get the screen by id
    ScreenResponseDto getScreenByid(Long id);
    //update the screen
    ScreenResponseDto updateScreen(Long id,ScreenRequestDto requestDto);
    //Delete the Screen permanently
    void DeleteScreen(Long id);
    //soft delete
    void deactivateScreen(Long id);
    //activate the screen
    ScreenResponseDto activateScreen(Long id);

    //search
    List<ScreenResponseDto>searchScreen(String screenName);
    //search by screen type
    List<ScreenResponseDto> searchByScreenType(String screenType);
    //search by only active secreen
    List<ScreenResponseDto> getActiveScreens();
    //get screen that is not active
    List<ScreenResponseDto>getNotActiveScreens();
    //count the total active screen
    long countActiveScreens();
    //Get the screen By Theater Id

    List<ScreenResponseDto>getScreenByTheater(Long theaterId);

}
